### DISCUSSION TITLE
high % of unspliced

### DISCUSSION THREAD

**Discussion by chrismahony** [theislab/scvelo#845] [Q&A] [answered]
When I run scVelo, I get high levels of unspliced RNAs.

This is how extracted details from my Seurat object:

eurat_obj$barcode <- colnames(seurat_obj)
seurat_obj$UMAP_1 <- seurat_obj@reductions$umap@cell.embeddings[,1]
seurat_obj$UMAP_2 <- seurat_obj@reductions$umap@cell.embeddings[,2]
write.csv(seurat_obj@meta.data, file='metadata.csv', quote=F, row.names=F)

library(Matrix)
counts_matrix <- GetAssayData(seurat_obj, assay='RNA', slot='counts')
writeMM(counts_matrix, file=paste0('counts.mtx'))
write.csv(seurat_obj@reductions$pca@cell.embeddings, file='pca.csv', quote=F, row.names=F)
write.table(
  data.frame('gene'=rownames(counts_matrix)),file='gene_names.csv',
  quote=F,row.names=F,col.names=F
)

And how I created my adata object:

X = io.mmread("/path/counts.mtx")

# create anndata object
adata = anndata.AnnData(
    X=X.transpose().tocsr()
)
# load cell metadata:
cell_meta = pd.read_csv("metadata.csv")

# load gene names:
with open("gene_names.csv", 'r') as f:
    gene_names = f.read().splitlines()

# set anndata observations and index obs by barcodes, var by gene names
adata.obs = cell_meta
adata.obs.index = adata.obs['barcode']
adata.var.index = gene_names

# load dimensional reduction:
pca = pd.read_csv("pca.csv")
pca.index = adata.obs.index

# set pca and umap
adata.obsm['X_pca'] = pca.to_numpy()
adata.obsm['X_umap'] = np.vstack((adata.obs['UMAP_1'].to_numpy(), adata.obs['UMAP_2'].to_numpy())).T

#read in loom data
scv.settings.verbosity = 3

scv.settings.set_figure_params('scvelo', facecolor='white', dpi=100, frameon=False)
cr.settings.verbosity = 2

ldata1 = scv.read('/path/CON_A.loom', cache=True) #repeated for all 12 samples

barcodes = [bc.split(':')[1] for bc in ldata1.obs.index.tolist()]
barcodes = [bc[0:len(bc)-1] + '_1' for bc in barcodes]
ldata1.obs.index = barcodes 
ldata1.var_names_make_unique()  #repeated for all 12 samples

ldata = ldata1.concatenate([ldata2, ldata3, ldata4, ldata5, ldata6, ldata7, ldata8, ldata9, ldata10, ldata11, ldata12])
adata = scv.utils.merge(adata, ldata)

scv.pl.proportions(adata, groupby='res_0.15_cmcleaned', save='proportions.pdf')

#When I run:
scv.pp.filter_and_normalize(adata)
scv.pp.moments(adata)

I get:

Normalized count data: X, spliced, unspliced.
WARNING: Did not modify X as it looks preprocessed already.
computing neighbors
    finished (0:00:17) --> added 
    'distances' and 'connectivities', weighted adjacency matrices (adata.obsp)
computing moments based on connectivities
    finished (0:00:07) --> added 
    'Ms' and 'Mu', moments of un/spliced abundances (adata.layers)

I have taken the unnormalized counts from the seurat obj, so I am a bit confused? I asusme this is the issue, but I am not sure where the normalization is happening?

Any help would be great thanks,
Chris

[scvelo_proportions_proportions.pdf](https://github.com/theislab/scvelo/files/8340654/scvelo_proportions_proportions.pdf)

**Comment by WeilerP** [comment, marked as answer]
@chrismahony, `scv.pl.proportions` is meant to run on the raw counts, _i.e._, not yet preprocessed (e.g. normalized, etc.).
